position_in_partition.hh

```diff
@@ -676,3 +676,41 @@ bool position_range::overlaps(const schema& s, position_in_partition_view start,
     position_in_partition::less_compare less(s);
     return !less(end, _start) && less(start, _end);
 }
+
+// Track position_in_partittion transitions
+// and validate monotonicity.
+class position_in_partition_tracker {
```

Comment:
Perhaps `_tracker` is not the right name. Don't have a better one though.

Reply:
Renamed in v2.